### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by LeonidShatilo**
### Task:
Based on your 2nd task, make an additional opportunity to create and delete cards (for cards I mean any array of objects with image and description, for example, `const cards = [{ id: 1, category: 'male', imgUrl: '', price: 100 }])`.

### Requirements:
- Rewrite all classes components to functional components with hooks.
- Each `<Card />` component could be deleted from catalogue by clicking on the cross button (❌) which can be placed somewhere on the top of each card. After clicking on the button the array of cards in your `<CardsContainer />` should be updated.
- `<CardsContainer />` component should also additionally render `<CardsCreationForm />` as its children. This form should contain all the required inputs for creating the card (find a way on how to use only 1 handler for all the inputs) and a submit button. By clicking on a submit button the array of cards in your `<CardsContainer />` should be updated with the new card data.
- All the required handlers for creating/deleting cards should be in the `<CardsContainer />` component and passed down to child components as props.

### CODE REVIEW

File: src/components/Cards/CardsCreationForm/CardsCreationForm.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      const input = inputData[key];
+
+      return (
+        <Input
+          key={key + index}
+          type={input.type}
+          value={input.value}
+          placeholder={input.placeholder}
+          required={input.required}
+          onChange={(event) => handleChange(event, key)}
```
Comment (broshkabro) [inline src/components/Cards/CardsCreationForm/CardsCreationForm.js:19]: I reckon that for this app it is better to catch all input values on submit. `onChange `event leads to unnecessary re-render every time a user types something.

**Review by broshkabro — CHANGES_REQUESTED** [review]
There is an issue when I click submit button
image

File: src/components/Cards/CardsContainer/CardsContainer.js [hunk 2]
```diff
@@ -1,33 +1,122 @@
...
-  componentDidMount() {
+  useEffect(() => {
     apiCall().then((data) => {
-      this.setState({ cards: data });
+      setCardData({ cards: data });
     });
+  }, []);
+
+  function deleteCard(id) {
+    const cards = [...cardData.cards];
```
[the remark this answers is not in the document]
Comment (LeonidShatilo) [reply src/components/Cards/CardsContainer/CardsContainer.js]: You're right! Removed.

File: src/components/Cards/CardsCreationForm/CardsCreationForm.js (hunk 1 above)
Comment (LeonidShatilo) [reply src/components/Cards/CardsCreationForm/CardsCreationForm.js]: Implemented.

**Comment by LeonidShatilo** [comment]
> There is an issue when I click submit button
> image

I'm fixed this issue.

**Pull request merged**